subsys/bluetooth/host/gatt.c

```diff
@@ -1438,6 +1438,27 @@ static void gatt_delayed_store_enqueue(uint8_t id, const bt_addr_le_t *peer_addr
 	}
 }
 
+static void do_delayed_store(uint8_t id, const bt_addr_le_t *peer_addr)
+{
+	struct ds_peer *el = gatt_delayed_store_find(id, peer_addr);
+
+	if (bt_addr_le_is_bonded(id, peer_addr)) {
```

Comment:
Does it make sense to "just store what you promised to store"? I don't think checking if the peer is bonded serves any purpose here. If the peer has become un-bonded, the write-back cache "delayed store" should have been updated accordingly.

```suggestion
	if (el) {
```